python/cudf/cudf/tests/testing/test_assert_index_equal.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.
+
+import pandas as pd
+import pytest
+
+import cudf
+from cudf.testing import assert_index_equal
+
+
+@pytest.fixture(params=["equiv", True, False])
+def exact(request):
+    """Argument for assert_index_equal"""
+    return request.param
+
+
+def test_range_index_and_int_index_eqaulity(
+    signed_integer_types_as_str, exact
+):
+    pidx1 = pd.RangeIndex(0, stop=5, step=1)
+    pidx2 = pd.Index([0, 1, 2, 3, 4])
+    idx1 = cudf.from_pandas(pidx1)
+    idx2 = cudf.Index([0, 1, 2, 3, 4], dtype=signed_integer_types_as_str)
+
+    kind = None
+    try:
+        pd.testing.assert_index_equal(pidx1, pidx2, exact=exact)
+    except BaseException as e:
+        kind = type(e)
+
+    if kind is not None:
+        with pytest.raises(kind):
+            assert_index_equal(idx1, idx2, exact=exact)
+    else:
+        assert_index_equal(idx1, idx2, exact=exact)
```

Comment:
Should we move this logic into a shared utility? `test_cudf_assert_utility_follows_pandas(pd_assert_fn, cudf_assert_fn, *args, **kwargs)`

Reply:
Sure good idea. Yeah this pattern is prevalent a lot here